File: android/src/main/java/com/azify/theme/Message.java
package com.azify.theme;

import android.util.Log;

import com.azify.utils.Theme;

import org.json.JSONObject;

import java.util.Objects;

public class Message {
  private final Theme theme;

  public Message() {
    this.theme = new Theme();
  }

  private String getMessage(JSONObject theme, String key, String defaultMessage) {
    try {
      if (!this.theme.exists(theme, key)) {
        return defaultMessage;
      }

      return theme.getString(key);
    } catch (Exception error) {
      Log.d("Aziface", Objects.requireNonNull(error.getMessage()));
      return defaultMessage;
    }
  }

  public String getMessage(String target, String key, String defaultMessage) {
    final JSONObject theme = this.theme.getTarget(target);
    return this.getMessage(theme, key, defaultMessage);
  }

  public String getMessage(String target, String parent, String key, String defaultMessage) {
    final JSONObject targetTheme = this.theme.getTarget(target);
    final JSONObject parentTheme = this.theme.getTarget(targetTheme, parent);
    return this.getMessage(parentTheme, key, defaultMessage);
  }
}

File: ios/Theme/Message.swift
//
//  Message.swift
//  Pods
//
//  Created by Daniel Sansão Araldi on 15/08/25.
//

public class Message {
  private let style: Style

  init() {
    self.style = Style()
  }

  private func getMessage(_ theme: NSDictionary?, key: String, defaultMessage: String) -> String {
    if !self.style.exists(theme, key: key) {
      return defaultMessage
    }

    return theme?[key] as? String ?? defaultMessage
  }

  public func getMessage(_ target: String, key: String, defaultMessage: String) -> String {
    let theme = self.style.getTarget(target)
    return self.getMessage(theme, key: key, defaultMessage: defaultMessage)
  }

  public func getMessage(_ target: String, parent: String, key: String, defaultMessage: String)
    -> String
  {
    let targetTheme = self.style.getTarget(target)
    let parentTheme = self.style.getTarget(targetTheme, key: parent)
    return self.getMessage(parentTheme, key: key, defaultMessage: defaultMessage)
  }
}
